app/models.py

```diff
@@ -388,6 +391,17 @@ def serialize(self, data=None):
             "countersignedAt": countersigned_at,
         }, **(data or {}))
 
+        if self.agreement_details and self.agreement_details.get('uploaderUserId'):
+            user = User.query.filter(
+                User.id == self.agreement_details.get('uploaderUserId')
+            ).first()
+
+            if user:
```

Comment:
There's actually a precedent for this: https://github.com/alphagov/digitalmarketplace-api/blob/master/app/models.py#L808

It will just not return the user's information if there's no user found.